m4opt/models/bandpass/bandpass.py

```diff
@@ -0,0 +1,202 @@
+    >>> transmission = 0.5 * np.exp(-(waves-7000)**2/500**2)
+    >>> bp = Bandpass.from_table(waves* u.Angstrom, transmission)
+    >>> bp(7000*u.Angstrom)
+    <Quantity 0.49979598>
+
+    Alternatively, if you have the optional module `tynt`, we can instantiate
+    from `tynt`'s list of supported filters.
+
+    --->from m4opt.models.bandpass import tynt_filters
+    --->filter_name = tynt_filters[11]
+    --->print(filter_name)
+    'Generic/Johnson.V'
+    --->bp_V = Bandpass.from_name(filter_name)
+    --->bp_V(5555*u.Angstrom)
+    <Quantity 0.76578238>
+
+    Finally, you can instantiate a Bandpass model from an
+    `astropy.modeling.models` subclass and its associated input arguments.
+    Note, however, Bandpass objects require wavelength inputs and dimensionless
+    outputs.
+
+    >>> from astropy.modeling.models import Box1D, Gaussian1D
+    >>> bp_box = Bandpass.from_model(Box1D, amplitude=0.5, \
+        x_0 = 600*u.nm, width = 200*u.nm)
+    >>> bp_box(5500*u.Angstrom)
+    <Quantity 0.5>
+
+    >>> bp_gauss = Bandpass.from_model(Gaussian1D, amplitude=0.5, \
+        mean=6000*u.Angstrom, stddev=500*u.Angstrom)
+    >>> bp_gauss(5500*u.Angstrom)
+    <Quantity 0.30326533>
+    """
+
+    @classmethod
+    def from_table(cls, points, lookup_table):
+        """
+        Initializes bandpass from user-defined table.
+
+        points : 1D array of `astropy.Quantity` (wavelength or frequency)
+        lookup_table : 1D array of bandpass response
+        """
+
+        unit_type = get_physical_type(points)
+        if unit_type != "length" and unit_type != "frequency":
+            raise AttributeError("Input array points does not have " +
+                                 "appropriate wavelength or frequency " +
+                                 "units.")
+
+        result = Tabular1D(points, lookup_table*u.dimensionless_unscaled)
+        result.input_units_equivalencies = Background.input_units_equivalencies
+        return result
+
+    @classmethod
+    def from_name(cls, filter_name):
+        """
+        Initializes bandpass from tynt package.
+        """
+        if has_tynt:
+            if filter_name not in tynt_filters:
+                raise ValueError("Invalid filter {0}. ".format(filter_name) +
+                                 "See 'm4opt.bandpass.tynt_filters' for " +
+                                 "available options.")
+
+            filter = FilterGenerator().reconstruct(filter_name)
+            result = Tabular1D(filter.wavelength,
+                               filter.transmittance*u.dimensionless_unscaled)
+            result.input_units_equivalencies = (Background.
+                                                input_units_equivalencies)
+            return result
+
+        else:
+            print("optional dependency 'tynt' is not installed. " +
+                  "Will return 'None'")
+            return None
+
+    @classmethod
+    def from_model(cls, modelclass, **kwargs):
```

Comment:
Initializing a model from a model? I don't get it...

Reply:
This was my attempt to satisfy the criteria "User builds arbitrary model from other 1D models (e.g. Gaussian, compound model, etc.)" from #55. This imitates `synphot.SourceSpectrum` by taking in a `Model` class, and checking whether it has the appropriate input/output units. It also prevents using an inappropriate model for the Bandpass.

Reply:
That's fine. But in that case, we don't need to write any code to handle that situation? Any arbitrary `Model1D` can be compounded with any other model. All we would have to do is document that the input must have spectral units, and the output must be dimensionless.

Reply:
`from_model()` here does work as intended. 

If you're talking about taking a generic model and multiplying it with, e.g., `ZodiacalBackground`, then yes, it works (as long as the units are right)

Reply:
Then I think that we can just drop the `from_model()` method --- and the LICENSE file.